src/components/Loading/Loading.tsx

```diff
@@ -1,19 +1,19 @@
-import React, {useContext, useEffect, useRef} from 'react';
+import React, {useEffect, useRef} from 'react';
 import {Loading as AppBridgeLoading} from '@shopify/app-bridge/actions';
-import {FrameContext} from '../Frame';
+import {useFrame} from '../../utilities/frame';
 import {useAppBridge} from '../../utilities/app-bridge';
 
 export interface Props {}
 
 function Loading() {
   const appBridgeLoading = useRef<AppBridgeLoading.Loading>();
   const appBridge = useAppBridge();
-  const frame = useContext(FrameContext);
+  const {startLoading, stopLoading} = useFrame();
 
   useEffect(
     () => {
-      if (appBridge == null && frame) {
-        frame.startLoading();
+      if (appBridge == null) {
+        startLoading();
       } else if (appBridge != null) {
```

Comment:
If frame is always defined, then we can get ride of the if here and just have `} else {`